bin/commands/indexDump.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+const
+  fs = require('fs'),
+  ColorOutput = require('./colorOutput'),
+  path = require('path'),
+  ndjson = require('ndjson'),
+  getSdk = require('./getSdk');
+
+//@TODO use 'recursive: true' with node.js 10
+function mkdirp (fullPath) {
+  const parts = fullPath.split(path.sep);
+
+  if (parts.length > 1) {
+    mkdirp(parts.slice(0, parts.length - 1).join('/'));
+  }
+
+  if (! fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+function writeData(stream, data) {
+  return new Promise(resolve => stream.write(data, () => resolve()));
+}
+
+async function dumpCollection (sdk, index, collection, batchSize, directoryPath) {
+  const
+    filename = `${directoryPath}/${index}--${collection}--data.jsonl`,
+    writeStream = fs.createWriteStream(filename),
+    waitWrite = new Promise(resolve => writeStream.on('finish', resolve)),
+    ndjsonStream = ndjson.serialize(),
+    options = {
+      scroll: '10m',
+      size: batchSize
+    };
+
+  ndjsonStream.on('data', line => writeStream.write(line));
+
+  await writeData(ndjsonStream, { index, collection });
+  let
+    promises = [],
+    results = await sdk.document.search(index, collection, {}, options);
+
+  do {
+    process.stdout.write(`  ${results.fetched}/${results.total} documents dumped`);
+    process.stdout.write('\r');
+
+    for (const hit of results.hits) {
+      promises.push(writeData(ndjsonStream, {
+        _id: hit._id,
+        body: hit._source
+      }));
+    }
+
+    await Promise.all(promises);
+  } while (results = await results.next());
+
+  ndjsonStream.end();
+  writeStream.end();
+
+  return waitWrite;
+}
+
+async function indexDump (sdk, cout, index, batchSize, directoryPath) {
+  console.log(cout.notice(`Dumping index ${index} in ${directoryPath} ...`));
```

Comment:
To prevent a silent regression, you might want to make that PR depend on https://github.com/kuzzleio/kuzzle/pull/1342 and remove all usages of `console` (after #1342, `cout` directly prints the arguments on the console, you don't need to do that yourself anymore)